File: 02-programacion-basica/05-bucles/java/soluciones/RetoNumerosHastaCero.java
import java.util.Scanner;

public class RetoNumerosHastaCero {
    public static void main(String[] args) {
        int cantidad = 0;
        int suma = 0;

        try (Scanner scanner = new Scanner(System.in)) {
            while (true) {
                System.out.print("Número (0 termina): ");
                int numero = scanner.nextInt();
                // El centinela controla el final sin formar parte de los datos.
                if (numero == 0) break;
                suma += numero;
                cantidad++;
            }
        }

        double promedio = cantidad == 0 ? 0 : (double) suma / cantidad;
        System.out.printf("Cantidad: %d | Suma: %d | Promedio: %.2f%n", cantidad, suma, promedio);
    }
}

File: 02-programacion-basica/05-bucles/scala/soluciones/RetoNumerosHastaCero.scala
import scala.io.StdIn.readLine

@main def retoNumerosHastaCero(): Unit =
  var cantidad = 0
  var suma = 0
  var continuar = true

  // El valor 0 funciona como centinela y no se incluye en los cálculos.
  while continuar do
    val numero = readLine("Número (0 termina): ").toInt
    if numero == 0 then continuar = false
    else
      suma += numero
      cantidad += 1

  val promedio = if cantidad == 0 then 0.0 else suma.toDouble / cantidad
  println(f"Cantidad: $cantidad | Suma: $suma | Promedio: $promedio%.2f")
